chain/jsonrpc/src/lib.rs

```diff
@@ -1210,11 +1216,138 @@ fn get_cors(cors_allowed_origins: &[String]) -> Cors {
+                Err(_) => Ok(HttpResponse::Ok().json(&"failure")),
+            }
+        };
+        return result.boxed();
+    }
+    let response = format!("can't parse {}", query.address);
+    let result = async move { Ok(HttpResponse::Ok().json(&response)) };
+    result.boxed()
+}
+
+#[derive(Deserialize)]
+struct ConnectToInfo {
+    seed: String,
+    address: String,
+}
+
+fn connect_to(
+    query: web::Query<ConnectToInfo>,
+    handler: web::Data<JsonRpcHandler>,
+) -> impl Future<Output = Result<HttpResponse, HttpError>> {
+    let seed = query.seed.clone();
+    let dec = bs58::decode(seed);
+    //SecretKey::ED25519(secret_key) => bs58::encode(&secret_key.0[..]).into_string(),
+    let vec = dec.into_vec().unwrap();
+    let mut d = [0u8; 32];
+    for i in 0..32 {
+        d[i] = vec[i];
+    }
+
+    let pk = PublicKey::ED25519(ED25519PublicKey(d));
+    let id = pk.into();
+
+    let result = async move {
+        if let Ok(addr) = query.address.parse() {
+            let peer_info = PeerInfo::new(id, addr);
+            return match handler
+                .peer_manager_addr
+                .clone()
+                .send(OutboundTcpConnect { peer_info })
+                .await
+            {
+                Ok(_) => Ok(HttpResponse::Ok().json(&"success")),
+                Err(_) => Ok(HttpResponse::Ok().json(&"failure")),
+            };
+        } else {
+            Ok(HttpResponse::Ok().json(&"can't parse"))
+        }
+    };
+    result.boxed()
+}
+
+#[derive(Deserialize)]
+struct SetMinProtocolVersionInfo {
+    pub min_protocol_version: u32,
+}
+
+fn set_min_protocol_version(
+    query: web::Query<SetMinProtocolVersionInfo>,
+    handler: web::Data<JsonRpcHandler>,
+) -> impl Future<Output = Result<HttpResponse, HttpError>> {
+    let result = async move {
+        match handler
+            .peer_manager_addr
+            .send(SetMinProtocolVersion { min_protocol_version: query.min_protocol_version })
+            .await
+        {
+            Ok(_) => Ok(HttpResponse::Ok().json("ok")),
+            Err(_) => Ok(HttpResponse::Ok().json(&"failure")),
+        }
+    };
+    result.boxed()
+}
+
+fn get_peer_id(
+    handler: web::Data<JsonRpcHandler>,
+) -> impl Future<Output = Result<HttpResponse, HttpError>> {
+    let result = async move {
+        match handler.peer_manager_addr.send(GetPeerId {}).await {
+            Ok(val) => Ok(HttpResponse::Ok().json(&val)),
+            Err(_) => Ok(HttpResponse::Ok().json(&"failure")),
+        }
+    };
+    result.boxed()
+}
+
+fn get_network_stats(
+    handler: web::Data<JsonRpcHandler>,
+) -> impl Future<Output = Result<HttpResponse, HttpError>> {
+    let result = async move {
+        match handler.peer_manager_addr.send(GetNetworkStats {}).await {
+            Ok(val) => Ok(HttpResponse::Ok().json(&val)),
+            Err(_) => Ok(HttpResponse::Ok().json(&"failure")),
+        }
+    };
+    result.boxed()
+}
+
```

Comment:
The implementation here confuses me. What's the reason for not following how other rpc endpoints are implemented? cc @frol

Reply:
We have a specific way of creating and handling endpoints, they should return structured errors and all primitives for jspnrpc are in jsonrpc-primitives, please follow that pattern if you introduce new endpoints.

Reply:
fixed